### PR TITLE
Split implementation into smaller modules. Remove core/lit-html distinction.

### LINKED ISSUE #434: Merge core.js and lit-html.js

### PR DESCRIPTION
**Pull request by justinfagnani**
This PR is much easier to review commit-by-commit. For the most part I've tried not to update any code, comments, lets-labels, etc., so that it's just code movement. Where I've moved tests I've sometimes created new suites though.

All tests should pass at every commit.

Fixes #434

### CODE REVIEW

File: src/test/lit-html_test.ts [hunk 1]
```diff
@@ -122,7 +121,7 @@ suite('lit-html', () => {
       assert.equal(fooDiv, barDiv);
     });
 
-    test(
+    test.skip(
```
Comment (sorvell) [inline src/test/lit-html_test.ts]: Let's bring this test back.

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import {terser} from 'rollup-plugin-terser';
+
+export default {
+  input: 'index.js',
+  output: {
+    file: 'index.bundled.js',
+    format: 'esm',
+  },
+  onwarn(warning) {
+    if (warning.code !== 'CIRCULAR_DEPENDENCY') {
```
Comment (justinfagnani) [inline rollup.config.js]: @rictic does Closure support circular dependencies?

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import filesize from 'rollup-plugin-filesize';
```
Comment (justinfagnani) [inline rollup.config.js]: I'll add a license

File: rollup.config.js (hunk 2 above)
Comment (rictic) [reply rollup.config.js]: It supports some simple cases I tested, but I don't know how firm their guarantees of correctness are. We might want to test it out

**Pull request merged**